Snapshot transfer from leader to follower fails when log_impl == FiledBasedLog

`SynchronousTests.testSnapshotSentToFollower()` fails with
```
15:06:51,469  [ERROR] RAFT: B: failed applying snapshot from A: java.lang.IllegalArgumentException
java.lang.IllegalArgumentException: null
	at org.jgroups.raft.filelog.FilePositionCache.createDeleteCopyFrom(FilePositionCache.java:197) ~[classes/:?]
	at org.jgroups.raft.filelog.LogEntryStorage.removeOld(LogEntryStorage.java:214) ~[classes/:?]
	at org.jgroups.protocols.raft.FileBasedLog.truncate(FileBasedLog.java:196) ~[classes/:?]
	at org.jgroups.raft.util.LogCache.truncate(LogCache.java:147) ~[classes/:?]
	at org.jgroups.protocols.raft.RAFT.doSnapshot(RAFT.java:990) ~[classes/:?]
	at org.jgroups.protocols.raft.Follower.handleInstallSnapshotRequest(Follower.java:41) [classes/:?]
	at org.jgroups.protocols.raft.RAFT.handleUpRequest(RAFT.java:747) [classes/:?]
	at org.jgroups.protocols.raft.RAFT.up(RAFT.java:600) [classes/:?]
	at org.jgroups.raft.testfwk.RaftNode.up(RaftNode.java:106) [classes/:?]
	at org.jgroups.raft.testfwk.RaftCluster.send(RaftCluster.java:57) [classes/:?]
	at org.jgroups.raft.testfwk.RaftCluster.send(RaftCluster.java:47) [classes/:?]
	at org.jgroups.raft.testfwk.RaftNode.down(RaftNode.java:100) [classes/:?]
	at org.jgroups.protocols.raft.RAFT.sendSnapshotTo(RAFT.java:1012) [classes/:?]
	at org.jgroups.protocols.raft.RAFT.sendAppendEntriesMessage(RAFT.java:896) [classes/:?]
	at org.jgroups.protocols.raft.RAFT.flushCommitTable(RAFT.java:633) [classes/:?]
	at org.jgroups.tests.SynchronousTests.testSnapshotSentToFollower(SynchronousTests.java:490) [classes/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124) [testng-6.14.3.jar:?]
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583) [testng-6.14.3.jar:?]
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719) [testng-6.14.3.jar:?]
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989) [testng-6.14.3.jar:?]
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125) [testng-6.14.3.jar:?]
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109) [testng-6.14.3.jar:?]
	at org.testng.TestRunner.privateRun(TestRunner.java:648) [testng-6.14.3.jar:?]
	at org.testng.TestRunner.run(TestRunner.java:505) [testng-6.14.3.jar:?]
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455) [testng-6.14.3.jar:?]
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450) [testng-6.14.3.jar:?]
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415) [testng-6.14.3.jar:?]
	at org.testng.SuiteRunner.run(SuiteRunner.java:364) [testng-6.14.3.jar:?]
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52) [testng-6.14.3.jar:?]
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84) [testng-6.14.3.jar:?]
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208) [testng-6.14.3.jar:?]
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137) [testng-6.14.3.jar:?]
	at org.testng.TestNG.runSuites(TestNG.java:1049) [testng-6.14.3.jar:?]
	at org.testng.TestNG.run(TestNG.java:1017) [testng-6.14.3.jar:?]
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66) [testng-rt.jar:?]
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109) [testng-rt.jar:?]

java.lang.AssertionError: RAFT.log=first=0 commit=5 last=0 term=20 (0 entries, max-size=1024), expected last=5
```

--- a/src/org/jgroups/raft/filelog/FilePositionCache.java
+++ b/src/org/jgroups/raft/filelog/FilePositionCache.java
@@ -194,16 +194,21 @@ public class FilePositionCache {
       }
       final int pageIndex = toPageIndex(cacheIndex);
       if (pageIndex >= positionPages.size()) {
-         throw new IllegalArgumentException();
+         // truncating to an index larger than last appended (example, snapshot transfer from leader -> follower)
+         // nothing to copy
+         return new FilePositionCache(logIndex);
       }
       final int pageOffset = toPageOffset(cacheIndex);
       final PositionPage positionPage = positionPages.get(pageIndex);
       if (positionPage == null) {
-         throw new IllegalArgumentException();
+         // truncating to an index larger than last appended (example, snapshot transfer from leader -> follower)
+         // nothing to copy
+         return new FilePositionCache(logIndex);
       }
       final long positionToDecrement = positionPage.get(pageOffset);
       if (positionToDecrement == EMPTY) {
-         // if empty, we do not need to copy anything.
+         // truncating to an index larger than last appended (example, snapshot transfer from leader -> follower)
+         // nothing to copy
          return new FilePositionCache(logIndex);
       }
       // copy from cacheIndex until lastNotEmptyIndex

